brainbox/behavior/training.py

```diff
@@ -0,0 +1,517 @@
+
+
+def get_training_status(trials, task_protocol, ephys_sess_dates, n_delay):
+    """
+    Compute training status of a subject from three consecutive training datasets
+
+    :param trials: dict containing trials objects from three consective training sessions
+    :type trials: Bunch
+    :param task_protocol: task protocol used for the three training session, can be 'training',
+    'biased' or 'ephys'
+    :type task_protocol: list of strings
+    :param ephys_sess_dates: dates of sessions conducted on ephys rig
+    :type ephys_sess_dates: list of strings
+    :param n_delay: number of sessions on ephys rig with delay before start > 15 min
+    :type n_delay: int
+    :returns:
+        - status - training status of subject
+        - info - Bunch containing performance metrics that decide training status e.g performance
+                 on easy trials, number of trials, psychometric fit parameters, reaction time
+    """
+
+    info = Bunch()
+    trials_all = concatenate_trials(trials)
+
+    # Case when all sessions are trainingChoiceWorld
+    if np.all(np.array(task_protocol) == 'training'):
+        signed_contrast = get_signed_contrast(trials_all)
+        (info.perf_easy, info.n_trials,
+         info.psych, info.rt) = compute_training_info(trials, trials_all)
+        if not np.any(signed_contrast == 0):
+            status = 'in training'
+        else:
+            if criterion_1b(info.psych, info.n_trials, info.perf_easy, info.rt):
+                status = 'trained 1b'
+            elif criterion_1a(info.psych, info.n_trials, info.perf_easy):
+                status = 'trained 1a'
+            else:
+                status = 'in training'
+
+        return status, info
+
+    # Case when there are < 3 biasedChoiceWorld sessions after reaching trained_1b criterion
+    if ~np.all(np.array(task_protocol) == 'training') and \
+            np.any(np.array(task_protocol) == 'training'):
+        status = 'trained 1b'
+        (info.perf_easy, info.n_trials,
+         info.psych, info.rt) = compute_training_info(trials, trials_all)
+
+        return status, info
+
+    # Case when there is biasedChoiceWorld or ephysChoiceWorld in last three sessions
+    if not np.any(np.array(task_protocol) == 'training'):
+
+        (info.perf_easy, info.n_trials,
+         info.psych_20, info.psych_80,
+         info.rt) = compute_bias_info(trials, trials_all)
+        # We are still on training rig and so all sessions should be biased
+        if len(ephys_sess_dates) == 0:
+            assert(np.all(np.array(task_protocol) == 'biased'))
+            if criterion_ephys(info.psych_20, info.psych_80, info.n_trials, info.perf_easy,
+                               info.rt):
+                status = 'ready4ephysrig'
+            else:
+                status = 'trained 1b'
+
+        elif len(ephys_sess_dates) < 3:
+            assert(np.all(np.array([date in trials for date in ephys_sess_dates])))

```

Comment:
A little clearer w/o numpy `assert all(date in trials for date in ephys_sess_dates)`